api/src/utils/nova/searchExecutor.ts

```diff
@@ -24,165 +24,242 @@ export class SearchExecutor {
-                        };
-                    },
-                    "Foundry details fetch failed",
-                ),
+                new Promise((resolve, reject) => {
+                    this.apiService
+                        .foundryDetails(searchQuery.foundryId)
+                        .then((response) => {
+                            if (response.foundryDetails) {
+                                promisesResult = {
+                                    foundryId: searchQuery.foundryId,
+                                };
+                                resolve();
+                            } else {
+                                reject(new Error("Foundry details fetch failed"));
+                            }
+                        })
+                        .catch((_) => {
+                            reject(new Error("Foundry details fetch failed"));
+                        });
+                }),
             );
         }
 
         if (searchQuery.slotIndex) {
             promises.push(
-                this.executeQuery(
-                    this.apiService.getSlotCommitment(searchQuery.slotIndex),
-                    (_) => {
-                        promisesResult = {
-                            slotIndex: String(searchQuery.slotIndex),
-                        };
-                    },
-                    "Slot commitment fetch failed",
-                ),
+                new Promise((resolve, reject) => {
+                    this.apiService
+                        .getSlotCommitment(searchQuery.slotIndex)
+                        .then((response) => {
+                            if (searchQuery.slotIndex) {
+                                promisesResult = {
+                                    slotIndex: String(searchQuery.slotIndex),
+                                };
+                                resolve();
+                            } else {
+                                reject(new Error("Slot commitment fetch failed"));
+                            }
+                        })
+                        .catch((_) => {
+                            reject(new Error("Slot commitment fetch failed"));
+                        });
+                }),
             );
         }
 
         if (searchQuery.slotCommitmentId) {
             promises.push(
-                this.executeQuery(
-                    this.apiService.getCommitment(searchQuery.slotCommitmentId),
-                    (result) => {
-                        promisesResult = {
-                            slotIndex: String(result.slot.slot),
-                        };
-                    },
-                    "Slot commitment fetch failed",
-                ),
+                new Promise((resolve, reject) => {
+                    this.apiService
+                        .getCommitment(searchQuery.slotCommitmentId)
+                        .then((response) => {
+                            if (response?.slot?.slot) {
```

Comment:
Here i'm concerned with the case where `response.slot.slot === 0`, which could be a valid slot, but the condition will no enter (falsy). Maybe we can just be content with `if (response?.slot)` to be enough.